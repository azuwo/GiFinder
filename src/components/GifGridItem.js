import React from 'react'

const GifGridItem = ({title,url}) => {
    return (
        <div className = 'card'>
            <img src={url} alt={title}/>
        </div>
    )
}

export default GifGridItem
