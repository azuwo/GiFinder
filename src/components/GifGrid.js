import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({category}) => {

    const {data:images} = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
           { <div className = 'card-grid'>
                {
                    images.map( img  =>(
                        <GifGridItem
                            key= {img.id}
                            { ...img }
                        />
                    ))
                }
            </div>}
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
