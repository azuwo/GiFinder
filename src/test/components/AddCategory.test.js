import React from 'react'
import {shallow} from 'enzyme'
import {AddCategory} from "../../components/AddCategory"
import '@testing-library/jest-dom'


describe('test in AddCategory', () => {

    const setCategories = jest.fn()
    let wrapper

    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    })

    test('should show component correclty', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('should change textBox', () => {
        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', {target:{value}})
    })

    test('shouldnt change on submit ', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories and clean textBox', () => {
        //simulate input change
        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', {target:{value}})

        //simulate submit
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        //setCategories should have been called
        expect(setCategories).toHaveBeenCalled()
        //expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

        //input value = ''
        expect(input.prop('value')).toBe('')

    });
})