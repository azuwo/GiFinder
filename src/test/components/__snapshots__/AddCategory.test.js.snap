// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in AddCategory should show component correclty 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
